front/pages/w/[wId]/a/index.tsx

```diff
@@ -469,13 +482,98 @@ export function Providers({ owner }: { owner: WorkspaceType }) {
+      <Page.SectionHeader
+        title="Dust Apps"
+        description="Create and manage your custom Large Language Models apps."
+        action={{
+          label: "Create App",
+          variant: "primary",
+          onClick: async () => {
+            void router.push(`/w/${owner.sId}/a/new`);
+          },
+          icon: PlusIcon,
+        }}
+      />
+      <ul role="list" className="pt-4">
+        {apps.map((app) => (
+          <li key={app.sId} className="px-2">
+            <div className="py-4">
+              <div className="flex items-center justify-between">
+                <Link href={`/w/${owner.sId}/a/${app.sId}`} className="block">
+                  <p className="truncate text-base font-bold text-action-600">
+                    {app.name}
+                  </p>
+                </Link>
+                <div className="ml-2 flex flex-shrink-0">
+                  <p
+                    className={classNames(
+                      "inline-flex rounded-full px-2 text-xs font-semibold leading-5",
+                      app.visibility == "public"
+                        ? "bg-green-100 text-green-800"
+                        : "bg-gray-100 text-gray-800"
+                    )}
+                  >
+                    {app.visibility}
+                  </p>
+                </div>
+              </div>
+              <div className="mt-2 sm:flex sm:justify-between">
+                <div className="sm:flex">
+                  <p className="flex items-center text-sm text-gray-700">
+                    {app.description}
+                  </p>
+                </div>
+                <div className="mt-2 flex items-center text-sm text-gray-300 sm:mt-0">
+                  <p>{app.sId}</p>
+                </div>
+              </div>
+            </div>
+          </li>
+        ))}
+        {apps.length == 0 ? (
+          <div className="mt-6 flex flex-col items-center justify-center text-sm text-gray-500">
+            <p>Welcome to the Dust developer platform 🔥</p>
+            <p className="mt-2">
+              Setup your Providers (below) or create your first app to get
+              started.
+            </p>
+            <p className="mt-6">
+              You can also visit our developer documentation:
+            </p>
+            <p className="mt-2">
+              <Link href="https://docs.dust.tt" target="_blank">
+                <Button variant="tertiary" label="View Documentation" />
+              </Link>
+            </p>
+          </div>
+        ) : null}
+      </ul>
+    </Page.Vertical>
+  );
+}
+
 export default function Developers({
   owner,
   subscription,
   apps,
   gaTrackingId,
 }: InferGetServerSidePropsType<typeof getServerSideProps>) {
+  const [currentTab, setCurrentTab] = useState("apps");
   const router = useRouter();
+  const handleTabChange = async (tabId: string) => {
+    const query = { ...router.query, t: tabId };
+    await router.push({ query });
```

Comment:
You might want to use `replace` here to keep a clean history.

Reply:
🙏 great, will bundle in the next dustapps-ux PR